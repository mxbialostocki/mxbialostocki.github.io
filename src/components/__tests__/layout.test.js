import React from "react"
import ShallowRenderer from "react-test-renderer/shallow"

import Layout from '../layout'
import { useStaticQuery } from "gatsby"

// const { useStaticQuery } = jest.requireMock("gatsby")

beforeAll(() => {
  useStaticQuery.mockReturnValue({
    site: {
      siteMetadata: {
        title: "Mock Title",
        description: "Mock Description"
      }
    }
  })
})

describe("Layout", () => {
  const renderer = new ShallowRenderer()
  const children = 'children'  
  it("renders correctly", () => {
    const tree = renderer.render(
      <Layout>{children}</Layout>
    )
    expect(tree).toMatchSnapshot()
  })
})