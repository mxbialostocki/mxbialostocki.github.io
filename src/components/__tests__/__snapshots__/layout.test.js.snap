// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout renders correctly 1`] = `
<React.Fragment>
  <Header
    siteTitle="Mock Title"
  />
  <WithStyles(ForwardRef(Grid))
    className="makeStyles-layout-2"
    container={true}
    direction="row"
    justify="space-between"
  >
    <WithStyles(ForwardRef(Grid))
      item={true}
      xs={12}
    />
    <WithStyles(ForwardRef(Grid))
      item={true}
      xs={12}
    >
      <main>
        children
      </main>
    </WithStyles(ForwardRef(Grid))>
    <WithStyles(ForwardRef(Grid))
      item={true}
      xs={12}
    />
  </WithStyles(ForwardRef(Grid))>
  <Footer />
</React.Fragment>
`;

exports[`layout Renders correctly 1`] = `
<React.Fragment>
  <Header
    siteTitle="Mock Title"
  />
  <WithStyles(ForwardRef(Grid))
    className="makeStyles-layout-2"
    container={true}
    direction="row"
    justify="space-between"
  >
    <WithStyles(ForwardRef(Grid))
      item={true}
      xs={12}
    />
    <WithStyles(ForwardRef(Grid))
      item={true}
      xs={12}
    >
      <main>
        children
      </main>
    </WithStyles(ForwardRef(Grid))>
    <WithStyles(ForwardRef(Grid))
      item={true}
      xs={12}
    />
  </WithStyles(ForwardRef(Grid))>
  <Footer />
</React.Fragment>
`;

exports[`layout renders correctly 1`] = `
<React.Fragment>
  <Header
    siteTitle="Mock Title"
  />
  <WithStyles(ForwardRef(Grid))
    className="makeStyles-layout-2"
    container={true}
    direction="row"
    justify="space-between"
  >
    <WithStyles(ForwardRef(Grid))
      item={true}
      xs={12}
    />
    <WithStyles(ForwardRef(Grid))
      item={true}
      xs={12}
    >
      <main>
        children
      </main>
    </WithStyles(ForwardRef(Grid))>
    <WithStyles(ForwardRef(Grid))
      item={true}
      xs={12}
    />
  </WithStyles(ForwardRef(Grid))>
  <Footer />
</React.Fragment>
`;

exports[`layout renders footer 1`] = `
<React.Fragment>
  <Header
    siteTitle="Mock Title"
  />
  <WithStyles(ForwardRef(Grid))
    className="makeStyles-layout-2"
    container={true}
    direction="row"
    justify="space-between"
  >
    <WithStyles(ForwardRef(Grid))
      item={true}
      xs={12}
    />
    <WithStyles(ForwardRef(Grid))
      item={true}
      xs={12}
    >
      <main>
        children
      </main>
    </WithStyles(ForwardRef(Grid))>
    <WithStyles(ForwardRef(Grid))
      item={true}
      xs={12}
    />
  </WithStyles(ForwardRef(Grid))>
  <Footer />
</React.Fragment>
`;
